tests/unit/test_jobs.py

```diff
@@ -0,0 +1,33 @@
+from catkin_tools.jobs.utils import merge_envs
+
+
+def test_merge_envs_basic():
+    job_env = { 'PATH': '/usr/local/bin:/usr/bin', 'FOO': 'foo' }
+
+    merge_envs(job_env, [
+        { 'PATH': '/usr/local/bin:/bar/baz/bin' },
+        { 'BAR': 'bar' } ])
+
+    # Validate that the known path was not moved from the existing order, and the unfamiliar
+    # path was correctly prepended.
+    assert job_env['PATH'] == '/bar/baz/bin:/usr/local/bin:/usr/bin'
```

Comment:
I understand your logic, but what I'm talking about is expectation.

If I put myself in the shoes of the person setting `/usr/local/bin:/bar/baz/bin`, I might expect that the merged path might contain other stuff in between my entries, e.g. `X:/usr/local/bin:Y:/bar/baz/bin:Z`, but I would not expect my entries to end up reordered, where the first occurrence of `/bar/baz/bin` comes up before the first occurrence of `/usr/local/bin`.

I think that "append only if not in list" makes sense, and I think "remove duplicates in list and then prepend" makes sense. But what you've got here is "prepend only if not in list" which doesn't make sense to me because it can cause the order of contributed entries to switch.